src/state.cpp

```diff
@@ -148,6 +148,44 @@ void pop_all()
     TRACE_FUNC_END;
 }
 
+bool check_states(const states::StateId state_given, const states::StateId state_to_check)
```

Comment:
Sorry but this function seems quite bizarre to me - I have no idea what the point is.